src/server/pmix_server_get.c

```diff
@@ -689,7 +689,7 @@ static pmix_status_t _satisfy_request(pmix_nspace_t *nptr, pmix_rank_t rank,
 
     /* retrieve the data for the specific rank they are asking about */
     if (PMIX_RANK_WILDCARD != rank) {
-        if (!peer->commit_cnt) {
+        if (!PMIX_PROC_IS_SERVER(peer) && !peer->commit_cnt) {
```

Comment:
I know you guys added this commit_cnt for a reason, but I honestly don't remember it any more. Can you add a brief comment explaining why it is needed? Or is that comment somewhere else in the code?

Reply:
@rhc54 `commit_cnt` was introduced in PR #613 that fixes issue #586. 
This works for local requests where `peer` is corresponds to client, 
but for remote requests `peer` corresponds to server peer: https://github.com/pmix/pmix/blob/master/src/server/pmix_server_get.c#L805-L810:
```
        scd = PMIX_NEW(pmix_server_caddy_t);
        PMIX_RETAIN(pmix_globals.mypeer);
        scd->peer = pmix_globals.mypeer;
        PMIX_LIST_FOREACH(req, &ptr->loc_reqs, pmix_dmdx_request_t) {
            pmix_status_t rc;
            rc = _satisfy_request(nptr, rank, scd, req->cbfunc, req->cbdata, NULL);
```
For server `commit_cnt` should not be taken into account, this always will be return [PMIX_ERR_NOT_FOUND](https://github.com/pmix/pmix/blob/1207023c664ffe5b6c3cd101fceef425944979b8/src/server/pmix_server_get.c#L693).

Reply:
Okay, let me see if I understand. The server is the one tracking this commit_cnt as it is the one who updates it. It updates that count whenever the client "commits" its data to the server. However, the server itself doesn't do the commit, and so the server's "peer" object will never have a non-zero commit_cnt.

The server stores committed data from its clients in its local storage. When someone requests that data, the server goes through the pmix_server_get code using its own peer object since that is where the data is stored. The current check prevented us from going thru the code path because it looked at the server's peer object, whose commit_cnt is always zero.

The correct check would have been to look at the commit_cnt of the proc whose info is being requested - yes? Wouldn't that be the correct fix here, instead of just ignoring the flag?

Reply:
@rhc54 Indeed this looks like a better solution.
@karasevb lets sync on this today.

Reply:
@rhc54 
I just talked to @karasevb and we think that this change is correct. Here is the reason.
* This portion originally worked ok if the key was requested from another local process.
* The issue was when we were applying this check to the case when the key was requested from the process on the remote node. In this case we simply don't have corresponding commit count as it is maintained by another PMIx server.

Reply:
Ah - thank you! Could you capture that explanation in a comment in an appropriate place in the code? I'm sure future developers working thru the code won't have that context.

Reply:
I'll add it to the commit message and in the code comment.

Reply:
@rhc54 , @artpol84
Code comment and commit message have been updated, PR ready to go.